src/api/python/CMakeLists.txt

```diff
@@ -43,18 +43,22 @@ endif()
 # Required for Cython target below
 list(APPEND CMAKE_MODULE_PATH ${SKBUILD_CMAKE_MODULE_PATH})
 
-# 'PythonExtensions' is not yet compatible with cmake's new 'FindPython'.
-#     https://github.com/scikit-build/scikit-build/issues/506
-# We define the following symbols below to be backward compatible.
-find_package(Python 3.6...3.10.999 COMPONENTS Development)
-
-# Since PythonExtensions still uses PythonInterp and PythonLibs, we have to
-# make sure that these modules find the same Python version that was found by
-# FindPython.
-find_package(PythonInterp ${Python_VERSION} REQUIRED)
-find_package(PythonLibs ${Python_VERSION} REQUIRED)
```

Comment:
What was the issue with these find_package calls?

Reply:
Line 46 could load a different version of python than the one loaded in cvc5/CMakeLists.txt. So I specified below the exact python version that needs to be loaded. 

The PythonInterp and PythonLibs were also loading conflicting python versions. So I got around this by defining backward compatible variables (PYTHON_EXECUTABLE, PYTHON_INCLUDE_DIRS and PYTHON_LIBRARIES).